group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -5419,6 +5429,13 @@ public void replay(
             streamsGroup.setGroupEpoch(value.epoch());
             streamsGroup.setMetadataHash(value.metadataHash());
             streamsGroup.setValidatedTopologyEpoch(value.validatedTopologyEpoch());
+
+            if (value.assignmentConfigs() != null) {
+                for (StreamsGroupMetadataValue.AssignmentConfig config : value.assignmentConfigs()) {
+                    streamsGroup.setAssignmentConfig(config.key(), config.value());
```

Comment:
I think this would never "unset" a configuration. For example, if we remove a configuration, we'd probably want to also remove it from the StreamsGroup object.  I would change setAssignmentConfig to setAssignmentConfigs and pass in a whole collection. Inside the method you can clear and putAll on the internal collection.